src/test/kotlin/io/bazel/kotlin/integration/KspCompileTest.kt

```diff
@@ -0,0 +1,244 @@
+package io.bazel.kotlin.integration
+
+import com.google.common.truth.Truth.assertWithMessage
+import io.bazel.kotlin.integration.RulesKotlinWorkspace.Companion.build
+import org.apache.commons.compress.archivers.tar.TarArchiveInputStream
+import org.junit.Test
+import java.nio.file.Files.copy
+import java.nio.file.Files.createTempDirectory
+import java.nio.file.Path
+import java.nio.file.StandardCopyOption
+import java.util.zip.GZIPInputStream
+import kotlin.io.path.createDirectories
+
+class KspCompileTest {
+
+  private fun unpackReleaseArchive(): Path {
+    return RulesKotlinWorkspace::class.java.classLoader.getResourceAsStream("_release.tgz")
+      ?.let { stream ->
+        createTempDirectory("rules_kotlin_release-")
+          .also { tmp ->
+            TarArchiveInputStream(GZIPInputStream(stream)).use { archive ->
+              generateSequence(archive::getNextEntry).forEach { entry ->
+                if (!entry.isDirectory) {
+                  copy(
+                    archive,
+                    tmp.resolve(entry.name).createDirectories(),
+                    StandardCopyOption.REPLACE_EXISTING,
+                  )
+                }
+              }
+            }
+          }
+      }
+      ?: error("Cannot find release repo")
+  }
+
+  @Test
+  fun `test ksp compile`() {
```

Comment:
We should really just switch to https://github.com/bazel-contrib/rules_bazel_integration_test so that we aren't having to write all of this Starlark in Kotlin like this.